### ISSUE TITLE
Support [jstype = JS_NUMBER] for int64

### ISSUE THREAD

**Issue by pzeinlinger** [bufbuild/protobuf-es#292]
protobuf-ts has an option to use Number to represent int64 fields which is often very convenient where int32 2^32 size is not enough to represent a value, but Javascript's MAX_SAFE_INTEGER (2^53) certainly is.

In my case, this happens when dealing with accounting and money.

**Comment by smaye81** [comment]
Hi, @pzeinlinger.  Sorry for the delay.  It was actually a conscious decision to not allow this.  I see your point, but it's obviously only effective if you can guarantee that no number in that field will ever exceed JavaScript's `MAX_SAFE_INTEGER`.  This may be the case in your scenario, but it could lead to others encountering subtle bugs.  Can you provide more detail on why an `int64` doesn't meet your needs?

**Comment by f1yingbanana** [comment]
I don't think this would cause subtle bugs, given that we require the field to be annotated to use 🤯
